src/sidecar/sidecarHandle.ts

```diff
@@ -298,15 +298,20 @@ export class SidecarHandle {
    */
   public async query<Result, Variables>(
     query: TadaDocumentNode<Result, Variables>,
-    connectionId: string,
+    connectionId?: string,
     // Mark second parameter as optional if Variables is an empty object type
     // The signature looks odd, but it's the only way to make optional param by condition
     ...[variables]: Variables extends Record<any, never> ? [never?] : [Variables]
   ): Promise<Result> {
-    const headers = new Headers({
-      ...this.defaultClientConfigParams.headers,
-      [SIDECAR_CONNECTION_ID_HEADER]: connectionId,
-    });
+    let headers: Headers;
+    if (connectionId) {
+      headers = new Headers({
+        ...this.defaultClientConfigParams.headers,
+        [SIDECAR_CONNECTION_ID_HEADER]: connectionId,
+      });
+    } else {
+      headers = new Headers(this.defaultClientConfigParams.headers);
+    }
```

Comment:
GQL queries for direct connections won't provide a connection ID https://github.com/confluentinc/vscode/blob/djs/direct-connection-resources/src/graphql/direct.ts#L16

Reply:
👍